test/conformance/walk.go

```diff
@@ -130,37 +296,67 @@ func scandir(dir string) {
 	}
 }
 
-func scanfile(path string) {
+func scanfile(path string, src interface{}) []conformanceData {
 	v := newVisitor()
-	file, err := parser.ParseFile(v.FileSet, path, nil, 0)
+	file, err := parser.ParseFile(v.FileSet, path, src, parser.ParseComments)
 	if err != nil {
 		panic(err)
 	}
 
+	v.cMap = ast.NewCommentMap(v.FileSet, file, file.Comments)
+
 	ast.Walk(v, file)
+	return v.tests
 }
 
 func main() {
-	args := os.Args[1:]
-	if len(args) < 1 {
+	flag.Parse()
+
+	if len(flag.Args()) < 1 {
 		fmt.Fprintf(os.Stderr, "USAGE: %s <DIR or FILE> [...]\n", os.Args[0])
 		os.Exit(64)
 	}
 
-	for _, arg := range args {
+	var confFile *os.File
+	if *confDoc {
+		var err error
+		confFile, err = os.Create("conformance.md")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error creating file conformance.md: %s", err)
+			os.Exit(64)
+		}
+		defer confFile.Close()
+
+		// Note: this assumes that you're running from the root of the kube src repo
```

Comment:
Ditto above about hardcoded pathnames.  Also, this kind of logic belongs in the build system.  Bazel has a way for us to specify a genrule and say what the inputs and outputs are.

That said, I'm not sure we want/need to check in the output markdown.  We are trying to minimize the amount of generated code that we check in.

Reply:
@mml we are not generating  cf_header.md but it is the header part of conformance document with general information about documenting conformance tests. This is static text.